common/src/main/java/io/novafoundation/nova/common/data/secrets/v2/SecretStoreV2.kt

```diff
@@ -122,6 +122,16 @@ suspend fun SecretStoreV2.getAccountSecrets(
     }
 }
 
+fun AccountSecrets.ethereumKeyPair(): EncodableStruct<KeyPairSchema>? = fold(
+    left = { it[MetaAccountSecrets.EthereumKeypair] },
+    right = { it[ChainAccountSecrets.Keypair] }
+)
```

Comment:
A potential problem here: It's not guaranteed that chain secrets is ethereum based and for this case chain secrets would be not null

Reply:
What do you think?

Reply:
I think we should avoid that. I suggest to explicitly use `MetaAccountSecrets.EthereumKeypair` when we are looking for base secrets and `ChainAccountSecrets.Keypair` when we are looking for a specific chainId. In that case with known chain id we will be calling just `getChainSecretsKeyPair()` or something alike, so no hidden assumptions around specific keypair format on SecretStore level - only the upper level will assume so